core/cloudflow-streamlets/src/main/scala/cloudflow/streamlets/StreamletPort.scala

```diff
@@ -76,6 +80,17 @@ trait CodecInlet[T] extends Inlet {
    * If no unique group Id is set (which is the default), streamlet instances will each receive part of the data (on this inlet).
    */
   def withUniqueGroupId: CodecInlet[T]
+
+  /**
+   * Sets a value for error handler for potential data unmarshalling errors
+   * If no error handler is specified, defaults to logging error and skipping record.
+   */
+  def withErrorHandler(f: (Array[Byte], Try[T]) => Option[T]): CodecInlet[T]
+
+  /**
+   * handle marshalling errors
+   */
+  def handleErrors(message: Array[Byte], result: Try[T]): Option[T]
```

Comment:
Does this have to be in the trait? Maybe add errorHandler here as def and implement as val in ProtoInlet and AvroInlet.

Reply:
changed to `val handleErrors: function`